Improve hooks documentation (spellcheck and position) (#3261)
odoo/documentation: content/developer/reference/frontend/hooks.rst

          </t>
        </div>
      `;

.. _frontend/hooks/useSpellCheck:

useSpellCheck
=============

Location
--------

`@web/core/utils/hooks`

Description
-----------

> Activate the spellcheck state to an input or textarea on focus by a t-ref="spellcheck" in

Reviewer: I would use more code quote in this section
```suggestion
Activate the spellcheck state to an input or textarea on focus by a `t-ref="spellcheck"` in
```

the current component. This state is then removed on blur, as well as the red outline, which
improves readability of the content.

The hook can also be used on any HTML element with the `contenteditable` attribute. To disable
spellcheck completely on elements that might be enabled by the hook, set explicitly the
`spellcheck` attribute as `false` on the element.

> In the following example, the spellcheck will be enabled on the first input, the textarea and
> the div with `contenteditable="true"`.
> 
> .. code-block:: javascript
> 
>     import { useSpellCheck } from "@web/core/utils/hooks";
> 
>     class Comp {
>       setup() {
>         this.simpleRef = useSpellCheck();
>         this.customRef = useSpellCheck({ refName: "custom" });
>         this.nodeRef = useSpellCheck({ refName: "container" });
>       }
>       static template = "Comp";
>     }
> 
> .. code-block:: xml
> 
>     <t t-name="Comp" owl="1">
>       <input t-ref="spellcheck" type="text"/>
>       <textarea t-ref="custom"/>
>       <div t-ref="container">
>         <input type="text" spellcheck="false"/>
>         <div contenteditable="true"/>
>       </div>
>     </t>

Reviewer 2: Use the `example` directive to highlight and clearly separate examples from the rest of the reference doc.
```suggestion
.. example::

   In the following example, the spellcheck will be enabled on the first input, the textarea and
   the div with `contenteditable="true"`.
   
   .. code-block:: javascript
   
       import { useSpellCheck } from "@web/core/utils/hooks";
       class Comp {
         setup() {
           this.simpleRef = useSpellCheck();
           this.customRef = useSpellCheck({ refName: "custom" });
           this.nodeRef = useSpellCheck({ refName: "container" });
         }
         static template = "Comp";
       }
   
   .. code-block:: xml
   
       <t t-name="Comp" owl="1">
         <input t-ref="spellcheck" type="text"/>
         <textarea t-ref="custom"/>
         <div t-ref="container">
           <input type="text" spellcheck="false"/>
           <div contenteditable="true"/>
         </div>
       </t>
```

API
---

.. js:function:: useSpellCheck([options])

    :param Options options: the spellcheck options (see table below)

.. list-table::
   :widths: 20 20 60
   :header-rows: 1

   * - Option
     - Type
     - Description
   * - `refName`
     - string

>      - this is a `useRef reference <https://github.com/odoo/owl/blob/master/doc/reference/hooks.md#useref>`_ for the element that will be

Reviewer: I don't know what it's worth, but there is an alias for the Owl GitHub now
```suggestion
     - this is a `useRef reference <{OWL_PATH}/doc/reference/hooks.md#useref>`_ for the element that will be
```
